packages/edit-site/src/components/global-styles/variations-typography.js

```diff
@@ -0,0 +1,219 @@
+	const context = useMemo( () => {
+		return {
+			user: {
+				settings: variation.settings ?? {},
+				styles: variation.styles ?? {},
+			},
+			base,
+			merged: mergeBaseAndUserConfigs( base, variation ),
+			setUserConfig: () => {},
+		};
+	}, [ variation, base ] );
+
+	const selectVariation = () => {
+		setUserConfig( () => {
+			return {
+				settings: variation.settings,
+				styles: variation.styles,
+			};
+		} );
+	};
+
+	const selectOnEnter = ( event ) => {
+		if ( event.keyCode === ENTER ) {
+			event.preventDefault();
+			selectVariation();
+		}
+	};
+
+	const isActive = useMemo( () => {
+		return areGlobalStyleConfigsEqual( user, variation );
+	}, [ user, variation ] );
+
+	let label = variation?.title;
+	if ( variation?.description ) {
+		label = sprintf(
+			/* translators: %1$s: variation title. %2$s variation description. */
+			__( '%1$s (%2$s)' ),
+			variation?.title,
+			variation?.description
+		);
+	}
+
+	const [ bodyFontFamilies, headingFontFamilies ] = getFontFamilies(
+		mergeBaseAndUserConfigs( base, variation )
+	);
+
+	const bodyPreviewStyle = getFamilyPreviewStyle( bodyFontFamilies );
+	const headingPreviewStyle = {
+		...getFamilyPreviewStyle( headingFontFamilies ),
+		fontSize: '1.2rem',
+	};
+
+	return (
+		<GlobalStylesContext.Provider value={ context }>
+			<div
+				className={ classnames(
+					'edit-site-global-styles-variations_item',
+					{
+						'is-active': isActive,
+					}
+				) }
+				role="button"
+				onClick={ selectVariation }
+				onKeyDown={ selectOnEnter }
+				tabIndex="0"
+				aria-label={ label }
+				aria-current={ isActive }
+				onFocus={ () => setIsFocused( true ) }
+				onBlur={ () => setIsFocused( false ) }
+			>
+				<VStack
+					className="edit-site-global-styles-variations_item-preview"
+					isFocused={ isFocused }
+					style={ {
+						height: 70,
+						lineHeight: 1.2,
+						textAlign: 'center',
+					} }
+				>
+					<div style={ headingPreviewStyle }>
+						{ headingFontFamilies.name }
+					</div>
+					<div style={ bodyPreviewStyle }>
+						{ bodyFontFamilies.name }
+					</div>
+				</VStack>
+			</div>
+		</GlobalStylesContext.Provider>
+	);
+}
+
+export default function TypographyVariations() {
+	const variations = useSelect( ( select ) => {
+		return select(
+			coreStore
+		).__experimentalGetCurrentThemeGlobalStylesVariations();
+	}, [] );
+
+	const { base, user } = useContext( GlobalStylesContext );
+
+	const typographyVariations =
+		variations && getVariationsByProperty( user, variations, 'typography' );
+
+	const uniqueTypographyVariations = [];
+	const uniqueTypographyNames = [];
+	const isDup = ( x, y ) => {
+		return uniqueTypographyNames.find( ( it ) => {
+			return JSON.stringify( it ) === JSON.stringify( [ x, y ] );
+		} );
+	};
+
+	typographyVariations?.forEach( ( variation ) => {
+		const [ bodyFontFamilyName, headingFontFamilyName ] =
+			getFontFamilyNames( mergeBaseAndUserConfigs( base, variation ) );
```

Comment:
Maybe we need to come up with a way to name style variations with no font-families, e.g., cases where variations have typography styles only like this:

```json
{
	"settings": {},
	"styles": {
		"typography": {
			"letterSpacing": "10px"
		},
		"blocks": {
			"core/post-title": {
				"typography": {
					"fontWeight": "700"
				},
				"color": {
					"background": "red"
				}
			}
		}
	}
}
```

Possibly `${variationTitle} - typeset` or, when `variation.title` doesn't exist in the theme.json `${variationFileName} - typeset`?

<img width="278" alt="Screenshot 2024-02-05 at 2 17 05 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/fa02c3b9-729d-4728-b053-69d67d7ab997">

Reply:
> Maybe we need to come up with a way to name style variations with no font-families, e.g., cases where variations have typography styles only like this:

To simplify this, I propose we lean in on a "Font book" type approach, where the heading and body fonts are used in for `A` and `a` respectively, like this: 

![CleanShot 2024-02-06 at 13 36 34](https://github.com/WordPress/gutenberg/assets/1813435/460ad2b2-3a37-4740-a8e5-f5fb76c126a9)

We already do that for styles: 

![CleanShot 2024-02-06 at 13 38 10](https://github.com/WordPress/gutenberg/assets/1813435/ede48aba-8233-4375-8032-23c3ab3edb86)

Reply:
I updated the PR to use these designs:

<img width="315" alt="Screenshot 2024-02-24 at 10 34 34" src="https://github.com/WordPress/gutenberg/assets/275961/5067db96-8bd8-4862-82c3-1e2c3da100c2">

Reply:
Yeah, this looks good. We can expand later on listing exactly what aspects are being modified, which is a general concern of variations.

Reply:
Looks great!